doc/source/servers/batch.md

```diff
@@ -0,0 +1,149 @@
+                                  format for the data in the input file which
+                                  can be an array
+
+  -w, --workers INTEGER           The number of parallel request processor
+                                  workers to run for parallel processing
+
+  -r, --retries INTEGER           The number of retries for each request
+                                  before marking an error
+
+  -i, --input-data-path PATH      The local filestore path where the input
+                                  file with the data to process is located
+
+  -o, --output-data-path PATH     The local filestore path where the output
+                                  file should be written with the outputs of
+                                  the batch processing
+
+  -m, --method [predict]          The method of the SeldonDeployment to send
+                                  the request to which currently only supports
+                                  the predict method
+
+  -l, --log-level [debug|info|warning|error]
+                                  The log level for the batch processor
+  -b, --benchmark                 If true the batch processor will print the
+                                  elapsed time taken to run the process
+
+  -u, --batch-id TEXT             Unique batch ID to identify all datapoints
+                                  processed in this batch, if not provided is
+                                  auto generated
+
+  --help                          Show this message and exit.
+
+```
+
+### Identifiers
+
+Each data point that is sent to the Seldon Core model contains the following identifiers in the request metadata:
+* Batch ID - A unique identifier which can be provided through CLI or is automatically generated
+* Batch Instance ID - A generated unique identifier for each datapoint processed
+* Batch Index - The local ordered descending index for the datapoint relative to the input file location
+
+These identifiers are added on each request as follows:
+
+```
+seldon_request = {
+    <data>: <current_batch_instance>,
+    "meta": {
+        "tags": {
+            "batch_id": <BATCH_ID>
+            "batch_instance_id": <BATCH_INSTANCE_ID>
+            "batch_index": <BATCH_INDEX>
+            }
+        }
+    }
+```
+
+This allows the requests to be identified and matched against the initial request in the data.
+
+### Performance
+
+The implementation of the module is done leveraging Python's Threading system. 
+
+Benchmarking was carried out using vanilla Python requests module to assess performance of Threading vs Twisted vs AsyncIO. The results showed better performance with Asyncio, however given that the logic in the worker is quite minimal (ie sending a request) and most of the time is waiting for the request, the implementation with Python's native threading was able to perform at speeds that were efficient enough to very easily scale to thousands of workers.
+
```

Comment:
"waiting for request" -> "waiting for response"?

Reply:
Done